runtimes/peregrine/src/kilt.rs

```diff
@@ -0,0 +1,357 @@
+	type NetworkRewardRate = constants::staking::NetworkRewardRate;
+	type NetworkRewardStart = constants::staking::NetworkRewardStart;
+	type NetworkRewardBeneficiary = SendDustAndFeesToTreasury<Runtime>;
+	type WeightInfo = weights::parachain_staking::WeightInfo<Runtime>;
+
+	const BLOCKS_PER_YEAR: BlockNumberFor<Self> = constants::BLOCKS_PER_YEAR;
+}
+
+impl pallet_inflation::Config for Runtime {
+	type Currency = Balances;
+	type InitialPeriodLength = constants::treasury::InitialPeriodLength;
+	type InitialPeriodReward = constants::treasury::InitialPeriodReward;
+	type Beneficiary = SendDustAndFeesToTreasury<Runtime>;
+	type WeightInfo = weights::pallet_inflation::WeightInfo<Runtime>;
+}
+
+impl public_credentials::Config for Runtime {
+	type RuntimeHoldReason = RuntimeHoldReason;
+	type AccessControl = PalletAuthorize<DelegationAc<Runtime>>;
+	type AttesterId = DidIdentifier;
+	type AuthorizationId = AuthorizationId<<Runtime as delegation::Config>::DelegationNodeId>;
+	type CredentialId = Hash;
+	type CredentialHash = BlakeTwo256;
+	type Currency = Balances;
+	type Deposit = constants::public_credentials::Deposit;
+	type EnsureOrigin = EnsureDidOrigin<DidIdentifier, AccountId>;
+	type MaxEncodedClaimsLength = constants::public_credentials::MaxEncodedClaimsLength;
+	type MaxSubjectIdLength = constants::public_credentials::MaxSubjectIdLength;
+	type OriginSuccess = DidRawOrigin<AccountId, DidIdentifier>;
+	type RuntimeEvent = RuntimeEvent;
+	type SubjectId = AssetDid;
+	type WeightInfo = weights::public_credentials::WeightInfo<Runtime>;
+	type BalanceMigrationManager = Migration;
+}
+
+parameter_types! {
+	pub CheckingAccount: AccountId = PolkadotXcm::check_account();
+}
+
+pub(crate) type KiltToEKiltSwitchPallet = pallet_asset_switch::Instance1;
+impl pallet_asset_switch::Config<KiltToEKiltSwitchPallet> for Runtime {
+	type AccountIdConverter = AccountId32ToAccountId32JunctionConverter;
+	type AssetTransactor = FungiblesAdapter<
+		Fungibles,
+		MatchesSwitchPairXcmFeeFungibleAsset<Runtime, KiltToEKiltSwitchPallet>,
+		LocationToAccountIdConverter,
+		AccountId,
+		NoChecking,
+		CheckingAccount,
+	>;
+	type FeeOrigin = EnsureRoot<AccountId>;
+	type LocalCurrency = Balances;
+	type PauseOrigin = EnsureRoot<AccountId>;
+	type RuntimeEvent = RuntimeEvent;
+	type SubmitterOrigin = EnsureSigned<AccountId>;
+	type SwitchHooks = RestrictSwitchDestinationToSelf;
+	type SwitchOrigin = EnsureRoot<AccountId>;
+	type UniversalLocation = UniversalLocation;
+	type WeightInfo = weights::pallet_asset_switch::WeightInfo<Runtime>;
+	type XcmRouter = XcmRouter;
+
+	#[cfg(feature = "runtime-benchmarks")]
+	type BenchmarkHelper = crate::benchmarks::asset_switch::CreateFungibleForAssetSwitchPool1;
+}
+
+impl pallet_migration::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Balances;
+	type MaxMigrationsPerPallet = constants::pallet_migration::MaxMigrationsPerPallet;
+	type WeightInfo = weights::pallet_migration::WeightInfo<Runtime>;
+}
```

Comment:
Lazy migration pallets should, maybe also live in the migration section.

Reply:
Moved it into the migration module, good suggestion: https://github.com/KILTprotocol/kilt-node/pull/797/commits/24c894ac807fd1d4867339b6afb69900accafcad.